src/ReactiveUI.XamForms/RoutedViewHost.cs

```diff
@@ -94,14 +99,19 @@ public RoutedViewHost()
                             finally
                             {
                                 currentlyPopping = false;
-                                ((IViewFor)CurrentPage).ViewModel = Router.GetCurrentViewModel();
+                                if (CurrentPage is IViewFor page && x.CurrentViewModel != null)
+                                {
+                                    page.ViewModel = x.CurrentViewModel;
+                                }
                             }
 
                             return Unit.Default;
                         })
-                    .Subscribe());
+                    .Subscribe()
```

Comment:
I think a `.Concat()` is needed here (https://gist.github.com/garyng/4f0a461eb7524b37b8a440d8bbff5a2a#file-routedviewex-cs-L142)? So that a new `NavigateBack` request will wait for the previous one to complete instead of running concurrently with each other (which is the main reason why the `NavigationStack` will go out of sync).

Reply:
Resolved.